WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java

```diff
@@ -272,29 +279,35 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
             int newWidth = Integer.parseInt(newValue.toString());
             AppPrefs.setImageOptimizeMaxSize(newWidth);
             setDetailListPreferenceValue(mImageMaxSizePref,
-                                         newValue.toString(),
-                                         getLabelForImageMaxSizeValue(AppPrefs.getImageOptimizeMaxSize()));
+                    newValue.toString(),
+                    getLabelForImageMaxSizeValue(AppPrefs.getImageOptimizeMaxSize()));
         } else if (preference == mImageQualityPref) {
             AppPrefs.setImageOptimizeQuality(Integer.parseInt(newValue.toString()));
             setDetailListPreferenceValue(mImageQualityPref,
-                                         newValue.toString(),
-                                         getLabelForImageQualityValue(AppPrefs.getImageOptimizeQuality()));
+                    newValue.toString(),
+                    getLabelForImageQualityValue(AppPrefs.getImageOptimizeQuality()));
         } else if (preference == mOptimizedVideo) {
             AppPrefs.setVideoOptimize((Boolean) newValue);
             mVideoEncorderBitratePref.setEnabled((Boolean) newValue);
         } else if (preference == mVideoWidthPref) {
             int newWidth = Integer.parseInt(newValue.toString());
             AppPrefs.setVideoOptimizeWidth(newWidth);
             setDetailListPreferenceValue(mVideoWidthPref,
-                                         newValue.toString(),
-                                         getLabelForVideoMaxWidthValue(AppPrefs.getVideoOptimizeWidth()));
+                    newValue.toString(),
+                    getLabelForVideoMaxWidthValue(AppPrefs.getVideoOptimizeWidth()));
         } else if (preference == mVideoEncorderBitratePref) {
             AppPrefs.setVideoOptimizeQuality(Integer.parseInt(newValue.toString()));
             setDetailListPreferenceValue(mVideoEncorderBitratePref,
-                                         newValue.toString(),
-                                         getLabelForVideoEncoderBitrateValue(AppPrefs.getVideoOptimizeQuality()));
+                    newValue.toString(),
+                    getLabelForVideoEncoderBitrateValue(AppPrefs.getVideoOptimizeQuality()));
         } else if (preference == mStripImageLocation) {
             AppPrefs.setStripImageLocation((Boolean) newValue);
+        } else if (preference == mAppThemePreference) {
+            AppThemeUtils.Companion.setAppTheme(getActivity(), (String) newValue);
+            // restart activity to make sure changes are applied to PreferenceScreen
+            Intent refresh = new Intent(getActivity(), getActivity().getClass());
+            startActivity(refresh);
+            getActivity().finish();
```

Comment:
We can override the `recreate()` method in the `AppSettingsActivity`  to make the transition between light and dark themes to be a cross-fade rather than jump cut.

```java
@Override
public void recreate() {
    Intent intent = new Intent(AppSettingsActivity.this, AppSettingsActivity.class);
    intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
    startActivity(intent);
    overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
    finish();
}
```

Then we would just call `getActivity().recreate();` from `AppSettingsFragment`.  What do you think?

Reply:
Sounds good! Crossfade looks really nice 👍